latch_cli/snakemake/workflow.py

```diff
@@ -0,0 +1,490 @@
+                    bindings=sorted(bindings, key=lambda b: b.var),
+                    upstream_nodes=upstream_nodes,
+                    flyte_entity=task,
+                )
+                node_map[job.jobid] = node
+
+        bindings = []
+        output_names = list(self.interface.outputs.keys())
+        for i, out in enumerate(output_names):
+
+            def find_upstream_node():
+                for j in self._dag.targetjobs:
+                    for depen, files in self._dag.dependencies[j].items():
+                        for f in files:
+                            if variable_name_for_file(f) == out:
+                                return depen.jobid, variable_name_for_value(
+                                    f, depen.output
+                                )
+
+            upstream_id, upstream_var = find_upstream_node()
+            promise_to_bind = Promise(
+                var=out,
+                val=NodeOutput(node=node_map[upstream_id], var=upstream_var),
+            )
+            t = self.python_interface.outputs[out]
+            b = binding_from_python(
+                out,
+                self.interface.outputs[out].type,
+                promise_to_bind,
+                t,
+            )
+            bindings.append(b)
+
+        self._nodes = list(node_map.values())
+        self._output_bindings = bindings
+
+    def execute(self, **kwargs):
+        return exception_scopes.user_entry_point(self._workflow_function)(**kwargs)
+
+
+T = TypeVar("T")
+
+
+class SnakemakeJobTask(PythonAutoContainerTask[T]):
+    def __init__(
+        self,
+        job: snakemake.jobs.Job,
+        inputs: Dict[str, Type],
+        outputs: Dict[str, Type],
+        target_file_for_input_param: Dict[str, str],
+        target_file_for_output_param: Dict[str, str],
+        is_target: bool,
+        interface: Interface,
+        task_type="python-task",
+    ):
+
+        name = f"{job.name}_{job.jobid}"
+
+        self.job = job
+        self._is_target = is_target
+        self._python_inputs = inputs
+        self._python_outputs = outputs
+        self._target_file_for_input_param = target_file_for_input_param
+        self._target_file_for_output_param = target_file_for_output_param
+
+        def placeholder():
+            ...
+
+        self._task_function = placeholder
+
+        super().__init__(
+            task_type=task_type,
+            name=name,
+            interface=interface,
+            task_config=None,
+            task_resolver=SnakemakeJobTaskResolver(),
+        )
+
+    def get_fn_code(self):
+
+        code_block = ""
+
+        fn_interface = f"\n\n@small_task\ndef {self.name}("
+        for idx, (param, t) in enumerate(self._python_inputs.items()):
+            fn_interface += f"{param}: {t.__name__}"
+            if idx == len(self._python_inputs) - 1:
+                fn_interface += ")"
+            else:
+                fn_interface += ", "
```

Comment:
```suggestion
        param_str = "(" + ", ".join(f"{param}: {t.__name__}" for param, t in self._python_inputs.items()) + ")"
```

Reply:
you need to add a comma behind every parameter unless its the last one, so you really need what i wrote